File: alternatingCharacters/alternatingCharactersDustin.cpp
#include <bits/stdc++.h>

using namespace std;

int alternatingCharacters(string input) 
{
    char lastCharacter = ' ';
    int deletedCharacters = 0;
    for (const auto & charater : input)
    {
        if(lastCharacter == ' ')
        {
            lastCharacter = charater;
            continue;
        }
        if(lastCharacter == charater)
        {
            deletedCharacters++;
        }
        else
        {
            lastCharacter = charater;
        }
    }
    return deletedCharacters;
}

int main()
{
    cout << alternatingCharacters("AAAA") << endl;

    cout << alternatingCharacters("BBBB") << endl;

    cout << alternatingCharacters("ABAB") << endl;

    cout << alternatingCharacters("AAABBB") << endl;

    return 0;
}

File: alternatingCharacters/alternatingCharactersDustin.py
#!/bin/python3

# Complete the alternatingCharacters function below.
def alternatingCharacters(string):
    lastCharater = ''
    deletedCharacters = 0
    for charater in string:
        if not lastCharater:
            lastCharater = charater
            continue
        if lastCharater == charater:
            deletedCharacters += 1
        else:
            lastCharater = charater
    return deletedCharacters
